TedTalks/TedTalks/Controllers/TedTalkListViewController.swift

```diff
@@ -52,4 +64,29 @@ extension TedTalkListViewController: UITableViewDelegate, UITableViewDataSource
         showTedTalkDetail(tedTalkManager.tedTalks[indexPath.row])
     }
     
+    //**************Ted talk Picker View
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return pickerData.count
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return pickerData[row]
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        self.selectedFilter = pickerData[row]
+        tedTalkManager.filter(keyword: tedTalksSearchBar.text ?? "", filter: self.selectedFilter)
+        //tableView.reloadData()
+    }
+    
+    //**************Search bar
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        tedTalkManager.filter(keyword: selectedFilter, filter: searchText)
+        //tableView.reloadData()
```

Comment:
You need to reload / update the table in here, otherwise the table will get out of sync with the model (because the actual list of talks changed but the UI was not updated) and cause an Index out of range crash.

Reply:
Fixed